devito/interfaces.py

```diff
@@ -270,13 +269,11 @@ def __init__(self, *args, **kwargs):
             self.indices = self._indices(**kwargs)
             self.dtype = kwargs.get('dtype', np.float32)
             self.space_order = kwargs.get('space_order', 1)
-            initializer = kwargs.get('initializer', None)
-            if initializer is not None:
-                assert(callable(initializer))
-            self.initializer = initializer
+            self.initializer = kwargs.get('initializer', None)
+            if self.initializer is not None:
+                assert(callable(self.initializer))
             self.numa = kwargs.get('numa', False)
-            self._data = kwargs.get('_data', None)
```

Comment:
You're dropping the `_data` backdoor here, which could be used to inject remotely allocated memory (eg. from Julia). I know we're not really using this right now, but let's keep in mind that we need to re-instate something similar if the Julia layer needs to provide it's own pointer, for example to provide non-copy data pointers.